flycheck.el

```diff
@@ -3196,12 +3213,14 @@ otherwise."
     (let ((file-name (flycheck-error-filename err))
           (message (flycheck-error-message err)))
       (and
-       ;; The error is relevant for the current buffer if it's got no file-name
-       ;; and the current buffer has no file name, too, or if it refers to the
-       ;; same file as the current buffer.
-       (or (and (not file-name) (not buffer-file-name))
-           (and buffer-file-name file-name
-                (flycheck-same-files-p file-name buffer-file-name)))
+       (or
+        ;; Neither the error nor buffer have a file name
+        (and (not file-name) (not buffer-file-name))
+        ;; Both have files, and they match
+        (and buffer-file-name file-name
+             (flycheck-same-files-p file-name buffer-file-name))
+        ;; This is a significant error from another file
+        (flycheck-relevant-error-other-file-p err))
```

Comment:
This logic feels a bit convoluted: `flycheck-relevant-error-other-file-p` is used in two places: here it simplifies to `(eq 'error (flycheck-error-level err))`, and… (see below)

Reply:
See comment below.

Reply:
OK, makes sense. Thanks!